tst/schedulers/bayesopt/gpautograd/test_kernel.py

```diff
@@ -134,37 +136,64 @@ def test_matern52_wrongshape():
     kernel = Matern52(dimension=3)
     kernel.collect_params().initialize()
     X1 = anp.random.normal(0.0, 1.0, (5, 2))
-    try: kmat = kernel(X1, X1)
-    except Exception as e: print(e)
-    try: kdiag = kernel.diagonal(X1)
-    except Exception as e: print(e)
+
+    with pytest.raises(Exception):
+        kernel(X1, X1)
+
+    with pytest.raises(Exception):
+        kernel.diagonal(X1)
+
     X2 = anp.random.normal(0.0, 1.0, (3, 3))
-    try: kmat = kernel(X2, X1)
-    except Exception as e: print(e)    
```

Comment:
I think this would not break the tests if the kernel would not raise an exception, you would only be missing a print statement. This would be easy to miss.

Reply:
Thanks!